### PR TITLE
Add to Cart API & Reducer Refactor

### LINKED ISSUE #77: Add item to Cart API route
POST user/:id/cart

### PR DESCRIPTION
**Pull request by csweeny**
Updates singleUser reducer and API routes to provide the ability to update item quantity independently of adding an item to a cart

resolves #77

### CODE REVIEW

File: client/store/singleUser.js [hunk 1]
```diff
@@ -65,18 +75,34 @@ export const fetchUser = (userId) => async (dispatch) => {
...
+      const user = getState().user;
+      let cartQuantity;
+      let method = 'POST';
+
+      for(let i = 0; i < user.cart.length; i++) {
+          if(user.cart[i].id === product.id) {
+            method = 'PUT'
+            // user.cart[i].quantity++
+            cartQuantity = user.cart[i].quantity + 1
+            dispatch(updateCartQuantity(product.id, cartQuantity))
```
Comment (gynnantonix) [inline client/store/singleUser.js]: Please call dispatch outside of the loop so it doesn't inadvertently get called multiple times

File: server/api/users.js [hunk 2]
```diff
@@ -70,39 +71,43 @@ router.get("/:userId", requireToken, async (req, res, next) => {
...
+      },
+    })
 
-      // add products into cart, setting quantity as we go
-      await OrderItem.bulkCreate(productList.map(p => ({
-        productId: p.id,
-        orderId: cart.id,
-        quantity: p.quantity
-      })))
+    await currentProd.update({productId: req.body.productId, quantity: req.body.quantity})
```
Comment (gynnantonix) [inline server/api/users.js:89]: what happens if no OrderItem is found?

Comment (Scutellicious) [reply server/api/users.js:89]: based on reducer logic, this shouldn't happen.  we can add more error handling in a different PR if we want

File: client/store/singleUser.js (hunk 1 above)
Comment (Scutellicious) [reply client/store/singleUser.js]: done

**Review by gynnantonix — APPROVED** [review]

**Pull request merged**